pkg/network/transport/object/grpc/replication.go

```diff
@@ -206,50 +205,24 @@ func objectFromMessage(gMsg *objectGRPC.Object) (*object.Object, error) {
 }
 
 func (s *Server) metaInfoSignature(o object.Object) ([]byte, error) {
-	firstObj, _ := o.FirstID()
-	if firstObj.IsZero() && o.HasParent() && o.SplitID() == nil {
-		// object itself is the first one
-		firstObj = o.GetID()
-	}
-
-	var deleted []oid.ID
-	var locked []oid.ID
-	switch o.Type() {
-	case object.TypeTombstone:
-		var t object.Tombstone
-		err := t.Unmarshal(o.Payload())
-		if err != nil {
-			return nil, fmt.Errorf("reading tombstoned objects: %w", err)
-		}
-
-		deleted = t.Members()
-	case object.TypeLock:
-		var l object.Lock
-		err := l.Unmarshal(o.Payload())
-		if err != nil {
-			return nil, fmt.Errorf("reading locked objects: %w", err)
-		}
-
-		locked = make([]oid.ID, l.NumberOfMembers())
-		l.ReadMembers(locked)
-	default:
-	}
-
 	currentBlock := s.nmState.CurrentBlock()
 	currentEpochDuration := s.nmState.CurrentEpochDuration()
 	firstBlock := (uint64(currentBlock)/currentEpochDuration + 1) * currentEpochDuration
 	secondBlock := firstBlock + currentEpochDuration
 	thirdBlock := secondBlock + currentEpochDuration
 
-	firstMeta := objectcore.EncodeReplicationMetaInfo(o.GetContainerID(), o.GetID(), firstObj, o.PayloadSize(), deleted, locked, firstBlock, s.mNumber)
-	secondMeta := objectcore.EncodeReplicationMetaInfo(o.GetContainerID(), o.GetID(), firstObj, o.PayloadSize(), deleted, locked, secondBlock, s.mNumber)
-	thirdMeta := objectcore.EncodeReplicationMetaInfo(o.GetContainerID(), o.GetID(), firstObj, o.PayloadSize(), deleted, locked, thirdBlock, s.mNumber)
+	firstMeta, err := objectcore.EncodeReplicationMetaInfo(o, firstBlock, s.mNumber)
+	if err != nil {
+		return nil, fmt.Errorf("encoding metadata")
+	}
+	secondMeta, _ := objectcore.EncodeReplicationMetaInfo(o, secondBlock, s.mNumber)
```

Comment:
one more proof of the refactoring dubiousness

Reply:
this is just a context of this func: you need to repeat the same encoding 3 times, error happens for all or does not happen at all

ok, it is easier to drop this commit